### PR TITLE
Avoid creating duplicated settings files when automigration is disabled

### PR DESCRIPTION
**Pull request by meaksh**
We introduced a PR https://github.com/cobbler/cobbler/pull/2966 to disable setting automigration to be executed by default, this works fine but we just noticed there are still duplicated settings files being created on each restart of cobbler or during instantiation of `CobblerAPI` object:

```console
suma-test-srv:~ # ls -ld /etc/cobbler/settings*
-rw-r--r-- 1 root   root    40 mar 15 06:55 /etc/cobbler/settings
-rw-r----- 1 root   root  4892 mar 15 01:44 /etc/cobbler/settings_20220315_01-44-38.yaml
-rw-r----- 1 root   root  5389 mar 15 01:45 /etc/cobbler/settings_20220315_01-45-13.yaml
-rw-r----- 1 root   root  5388 mar 15 01:46 /etc/cobbler/settings_20220315_01-46-12.yaml
-rw-r----- 1 root   root  5388 mar 15 03:47 /etc/cobbler/settings_20220315_03-47-21.yaml
-rw-r----- 1 root   root  5388 mar 15 03:48 /etc/cobbler/settings_20220315_03-48-21.yaml
-rw-r----- 1 root   root  5388 mar 15 06:55 /etc/cobbler/settings_20220315_06-55-46.yaml
-rw-r----- 1 root   root  5388 mar 15 06:58 /etc/cobbler/settings_20220315_06-58-10.yaml
-rw-r----- 1 root   root  5388 mar 15 09:03 /etc/cobbler/settings_20220315_09-03-42.yaml
-rw-r----- 1 root   root  5388 mar 15 11:38 /etc/cobbler/settings_20220315_11-38-08.yaml
-rw-r----- 1 root   root  5388 mar 15 11:38 /etc/cobbler/settings_20220315_11-38-45.yaml
-rw-r----- 1 root   root  5388 mar 15 11:40 /etc/cobbler/settings_20220315_11-40-07.yaml
-rw-r----- 1 root   root  5388 mar 15 11:40 /etc/cobbler/settings_20220315_11-40-25.yaml
-rw-r----- 1 root   root  5388 mar 15 11:42 /etc/cobbler/settings_20220315_11-42-10.yaml
-rw-r----- 1 root   root  5388 mar 15 12:09 /etc/cobbler/settings_20220315_12-09-19.yaml
-rw-r----- 1 wwwrun www   5388 mar 15 12:09 /etc/cobbler/settings_20220315_12-09-40.yaml
-rw-r----- 1 root   root  5388 mar 15 12:10 /etc/cobbler/settings_20220315_12-10-21.yaml
-rw-r----- 1 wwwrun www   5388 mar 15 12:10 /etc/cobbler/settings_20220315_12-10-27.yaml
-rw-r----- 1 wwwrun www   5388 mar 15 12:11 /etc/cobbler/settings_20220315_12-11-04.yaml
drwxr-x--- 2 root   www    119 mar 15 01:37 /etc/cobbler/settings.d
-rw-r----- 1 root   www   5388 mar 15 12:11 /etc/cobbler/settings.yaml
-rw-r--r-- 1 root   root 22573 mar 15 01:44 /etc/cobbler/settings.yaml.backup
-rw-r----- 1 root   root  5389 mar 15 01:45 /etc/cobbler/settings.yaml.bak


```

This PR simply fixed this issue by preventing we save a copy of the settings when automigration was not performed.

### CODE REVIEW

**Review by SchoolGuy — APPROVED** [review]
LGTM. Thanks for the fix. Sorry that I have overlooked this.

File: cobbler/api.py [hunk 1]
```diff
@@ -187,12 +187,15 @@ def __generate_settings(
...
                     "Settings are invalid and auto-migration is disabled. Please correct this manually!"
                 ) from error
+
         # Use normalized or migrated dict and create settings object
         new_settings = settings.Settings()
         new_settings.from_dict(normalized_settings)
 
-        # save to disk
-        new_settings.save(settings_path)
+        if yaml_dict.get("auto_migrate_settings", True):
```
[the remark this answers is not in the document]
Comment (SchoolGuy) [reply cobbler/api.py]: From my side this is fixed. Thus I will resolve this conversation.

**Comment by SchoolGuy** [comment]
@meaksh Could you squash the black commit into your commit where you introduced the changes please?

**Comment by meaksh** [comment]
@SchoolGuy done :+1:

**Pull request merged**